integration/token/fungible/views/accept.go

```diff
@@ -72,5 +73,24 @@ func (a *AcceptCashView) Call(context view.Context) (interface{}, error) {
 	assert.NoError(err, "failed to retrieve vault status for transaction [%s]", tx.ID())
 	assert.Equal(network.Valid, vc, "transaction [%s] should be in valid state", tx.ID())
 
+	// Check that the tokens are or are not in the vault
+	qe := vault.TokenVault().QueryEngine()
+	for _, output := range outputs.Outputs() {
+		tokenID := output.ID(tx.ID())
+		if output.Owner.Equal(id) {
+			// check it exists
+			_, toks, err := qe.GetTokens(tokenID)
+			assert.NoError(err, "failed to retrieve token [%s]", tokenID)
+			assert.Equal(1, len(toks), "expected one token")
+			assert.Equal(output.Quantity.Hex(), toks[0].Quantity, "token quantity mismatch")
+			assert.Equal(output.Type, toks[0].Type, "token type mismatch")
+		} else {
+			// check it does not exist
+			_, _, err := qe.GetTokens(tokenID)
+			assert.Error(err, "token [%s] should not exist", tokenID)
+			assert.True(strings.Contains(err.Error(), "token not found"))
+		}
+	}
+
```

Comment:
It seems that this code is added 3x in this PR. Please refactor

Reply:
fixed